libs/sdk-ui-loaders/src/dashboard/loadingStrategies/adaptiveComponentLoaders.ts

```diff
@@ -43,8 +43,13 @@ export const adaptiveDashboardEngineLoaderFactory =
  */
 export const adaptiveDashboardPluginLoaderFactory =
     (moduleFederationIntegration: ModuleFederationIntegration) =>
-    (ctx: DashboardContext, dashboard: IDashboardWithReferences): Promise<LoadedPlugin[]> => {
+    (
+        ctx: DashboardContext,
+        dashboard: IDashboardWithReferences,
+        options?: IDashboardPluginsLoaderOptions,
+    ): Promise<LoadedPlugin[]> => {
         if (!isEmpty(dashboard.references.plugins)) {
+            options!.beforePluginsLoaded({ externalPluginsCount: dashboard.references.plugins.length });
```

Comment:
Should we check that `options` is defined, just to be safe?

Reply:
fixed